src/gui/MainWindow.cc

```diff
@@ -2619,24 +2635,38 @@ void MainWindow::setSelection(int index)
     tabManager->open(QString::fromStdString(file));
   }
 
+  EditorInterface *editor = nullptr;
+  for (auto seditor : tabManager->editorList) {
+    auto editorPathName = seditor->filepath.toStdString();
+    auto locationPathName = location.filePath().generic_string();
+    if (editorPathName == locationPathName) {
+      editor = seditor;
+    }
+  }
+
+  if (editor == nullptr) {
+    return;
+  }
```

Comment:
This is probably why lookups into the current unsaved file doesn't work.

Reply:
I wonder if always having a file associated with an editor (in /tmp/xxqmfkqmfkjq.scad as long as it is not given a name) wouldn't simplify code flow.

Reply:
My last commit fix that.